2016/proposals/stolliker-ryan.md

```diff
@@ -0,0 +1,67 @@
+# Ryan Stolliker: Result Aggregation Server for Software Carpentry
+
+## Abstract
+
+During Software Carpentry's workshops which teach computing skills to the greater scientific community, learners use Python scripts to test that they have installed the programming environment and other dependencies correctly. Currently, this information is known only to the participants and cannot be used or tracked by the organizers. Thus, the goal of this project is to enable the collection of this installation data and to store it for future use and analysis.
+
+## Technical Details
+
+### Server
+
+All server management will be written in Python. The server itself will be implemented using [bottle](http://bottlepy.org/docs/dev/index.html), a simple web framework with functionality to parse forms and JSON requests and responses. SQLite will be used to create and manage the database which holds the results.
```

Comment:
@rstolliker one of the difficulties with smaller frameworks is database access and schema migration. Can you please include how do you want to handle these aspects?

I've suggested SQLAlchemy + Alembic for other student for this project, because they do work OK with Flask; I'm not sure how would Bottle work with SQLA (I never used Bottle).

Still, Django is perfectly fine for this project too.